language/move-core/types/src/transaction_argument.rs

```diff
@@ -13,6 +13,8 @@ pub enum TransactionArgument {
     Address(AccountAddress),
     U8Vector(#[serde(with = "serde_bytes")] Vec<u8>),
     Bool(bool),
+    // HACK
+    U8VectorVector(Vec<Vec<u8>>),
```

Comment:
CC @vgao1996 / @runtian-zhou for this part

Reply:
@matbd what is the `serde` specification on the `U8Vector` and does this need one for `U8VectorVector`?

Reply:
Yes, LCS/Serde support arbitrarily nested vectors. We will just need to add the proper translation code for `U8VectorVector`.

Reply:
@sblackshear @dariorussi To update the code generation, we need a final decision on this data structure. Special casing `U8Vector` (aka Bytes) is not entirely illogical but perhaps we'd rather avoid `U8VectorVector`. Also, because of `TypeTag`, we have recursive data structures in Libra types already.

Reply:
I'd be fine with us supporting constants in `TransactionArgument`s.
I agree with you that we could get away with a limited set of args if that was important.
I also agree with you that the recursion is fixable or we can work around it.
If we go with the more generic approach (all constants) we would not have to change `TransactionArgument`s for a long time (if not ever) and that has a big appeal.
If we want to limit `TransactionArgument`s then someone on the client side should make the call on what are the foreseeable arguments we want.